library/std/src/os/unix/net/ancillary.rs

```diff
@@ -369,7 +399,19 @@ impl<'a> Iterator for Messages<'a> {
     }
 }
 
-/// A Unix socket Ancillary data struct.
+/// A struct for accessing a Unix socket ancillary data packet.
+///
+/// An *ancillary data* (also known as *control data*) *packet* may be attached
+/// to a usual *data segment* transferred through a Unix socket.
+/// See [“2.10.11 Socket Receive Queue”](https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_10_11)
+/// in POSIX for details. An ancillary data packet is a list of *ancillary data objects*
+/// (also known as *control messages*).
+/// Here an ancillary data packet is represented by a [`SocketAncillary`] and a control message
+/// (see `struct cmsghdr` in the C API) is represented by an [`AncillaryData`].
+/// The C API for ancillary data:
+///
+/// - [`sys/socket.h`](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_socket.h.html)
+/// - [`cmsg`(3)](https://man7.org/linux/man-pages/man3/cmsg.3.html)
```

Comment:
Nice, documentation is one of the things that was sorely missing. But I wouldn't phrase it in terms of unix sockets. The underlying API can be used to attach metadata to any kind of socket. #82858 is going to extend it to udp sockets.

Reply:
Yes, this would documentation looks nice. After the merge, I will copy it for my PR #82858.